src/core.js

```diff
@@ -1300,12 +1300,12 @@ $.extend( $.validator, {
 			this.settings.messages[ element.name ].remote = previous.message;
 
 			param = typeof param === "string" && { url: param } || param;
-
-			if ( previous.old === value ) {
+			optionDataString = JSON.stringify( $.param( $.extend( { data: value }, param.data ) ) );
```

Comment:
I am fine with either `stringify` or `$.param`, no preferences on my side.

only problem was in the initial version which used both in combination. Its fine as is.